api/src/certification/evaluation/application/certification-rescoring-controller.js

```diff
@@ -1,14 +1,30 @@
+import { CertificationRescoringNotAllowedError } from '../../../shared/domain/errors.js';
 import { usecases } from '../../evaluation/domain/usecases/index.js';
+import * as courseAssessmentResultRepository from '../../session-management/infrastructure/repositories/course-assessment-result-repository.js';
 import { AlgorithmEngineVersion } from '../../shared/domain/models/AlgorithmEngineVersion.js';
 import * as certificationCourseRepository from '../../shared/infrastructure/repositories/certification-course-repository.js';
 import CertificationRescored from '../domain/events/CertificationRescored.js';
 
-const rescoreCertification = async function (request, h, dependencies = { certificationCourseRepository }) {
+const rescoreCertification = async function (
+  request,
+  h,
+  dependencies = { certificationCourseRepository, courseAssessmentResultRepository },
+) {
   const juryId = request.auth.credentials.userId;
   const certificationCourseId = request.params.certificationCourseId;
 
   const certificationCourse = await dependencies.certificationCourseRepository.get({ id: certificationCourseId });
 
+  const latestAssessmentResult = await dependencies.courseAssessmentResultRepository.getLatestAssessmentResult({
+    certificationCourseId,
+  });
+  if (
+    latestAssessmentResult &&
+    (latestAssessmentResult.status === 'cancelled' || latestAssessmentResult.status === 'rejected')
+  ) {
```

Comment:
```suggestion
  if (_isAssessmentResultNotRescorable(latestAssessmentResult)) {}
```

Nitpick: Pour faciliter un peu la lecture